packages/store/src/workspace/workspace.ts

```diff
@@ -15,76 +16,58 @@ export interface PageMeta {
   [key: string]: string | number | boolean;
 }
 
-class WorkspaceMeta extends Space {
-  private _workspace: Workspace;
+type WorkspaceMetaData = {
+  pages: Y.Array<unknown>;
+  versions: Y.Map<unknown>;
+  name: string;
+  avatar: string;
+};
+
+class WorkspaceMeta extends Space<WorkspaceMetaData> {
   private _prevPages = new Set<string>();
   pageAdded = new Signal<string>();
   pageRemoved = new Signal<string>();
   pagesUpdated = new Signal();
   commonFieldsUpdated = new Signal();
 
-  constructor(id: string, workspace: Workspace, awareness: Awareness) {
-    super(id, workspace.doc, awareness);
-    this._workspace = workspace;
-    this._yMetaRoot.observeDeep(this._handleEvents);
-  }
-
-  private get _yMetaRoot() {
-    return this.doc.getMap(this.id);
-  }
-
-  private get _yPages() {
-    if (!this._yMetaRoot.has('pages')) {
-      this._yMetaRoot.set('pages', new Y.Array());
-    }
-
-    return this._yMetaRoot.get('pages') as Y.Array<unknown>;
-  }
-
-  private get _yVersions() {
-    if (!this._yMetaRoot.has('versions')) {
-      this._yMetaRoot.set('versions', new Y.Map());
-    }
-
-    return this._yMetaRoot.get('versions') as Y.Map<unknown>;
-  }
-
-  private get _yName() {
-    if (!this._yMetaRoot.has('name')) {
-      return null;
-    }
-    return this._yMetaRoot.get('name') as string;
+  constructor(id: string, doc: BlockSuiteDoc, awareness: Awareness) {
+    super(id, doc, awareness, {
+      valueInitializer: {
+        pages: () => new Y.Array(),
+        versions: () => new Y.Map(),
+        avatar: () => '',
+        name: () => '',
+      },
+    });
+    this.origin.observeDeep(this._handleEvents);
   }
 
-  private get _yAvatar() {
-    if (!this._yMetaRoot.has('avatar')) {
-      return null;
-    }
-    return this._yMetaRoot.get('avatar') as Y.Text;
+  get pages() {
+    return this.proxy.pages;
   }
 
   get name() {
-    return this._yName ? this._yName.toString() : '';
+    return this.proxy.name;
   }
 
   get avatar() {
-    return this._yAvatar ? this._yAvatar.toString() : '';
+    return this.proxy.avatar;
   }
 
   setName(val: string) {
```

Comment:
```suggestion
  setName(name: string) {
```

suggest, rename variable name to `name`